src/pool/poolClient.ts

```diff
@@ -444,6 +464,49 @@ export class Client {
     this.exchangeRateTable[l1TokenAddress] = { ...this.exchangeRateTable[l1TokenAddress], ...exchangeRateTable };
     return this.exchangeRateTable[l1TokenAddress];
   }
+  async resolveStakingData(
+    l1TokenAddress: string,
+    userState: Awaited<ReturnType<UserState["read"]>>
+  ): Promise<StakeData> {
+    assert(this.config.acceleratingDistributorAddress, "Must have the accelerating distributor address");
+    const contract = this.getOrCreateAcceleratingDistributorContract();
+    const { cumulativeBalance } = await contract.getUserStake(l1TokenAddress, userState.address);
+
+    const stakeTxs = await contract.queryFilter(contract.filters.Stake(l1TokenAddress, userState.address));
+    const unstakeTxs = await contract.queryFilter(contract.filters.Unstake(l1TokenAddress, userState.address));
+
+    const stakes = [
+      ...stakeTxs.map((tx) => ({
+        amount: tx.args.amount,
+        block: BigNumber.from(tx.blockNumber),
+        hash: tx.transactionHash,
+        staker: tx.args.user,
+      })),
+      ...unstakeTxs.map((tx) => ({
+        amount: tx.args.amount.mul(-1),
+        block: BigNumber.from(tx.blockNumber),
+        hash: tx.transactionHash,
+        staker: tx.args.user,
+      })),
+    ];
+
+    // Logic: We have a list of staked transactions + a list of unstake
+    // transactions + the user's current balance. We know that if we add
+    // back all the unstake transactions to the cumulative staked balance,
+    // we'll have a value that represents the total amount of funds that
+    // has ever been staked on a user's account. If we then subtract out
+    // the total amount of funds the user has specifically staked from the
+    // external LP pool, we'll be left with exactly how much the user has
+    // claimed through an airdrop.
+    const amountAirdropped = stakes.reduce((acc, previous) => acc.add(previous.amount.mul(-1)), cumulativeBalance);
```

Comment:
this looked weird until i realized u set unstakes to -1, interesting way to do this.